hooks/use-siren-from-url.ts

```diff
@@ -0,0 +1,11 @@
+import { useRouter } from 'next/router';
+import { extractSirenOrSiretSlugFromUrl } from '#utils/helpers';
+
+export default function useSirenFromRouter(): string | null {
```

Comment:
Why use `sirenFrom` and not `pathFrom`? It seems to me that going redirecting to the exact page where the user started the auth workflow provides a better experience…

Reply:
I agree it is more powerful but it can becomes complicated for logout as the pathFrom is then forbidden 

I suggest we do this in a dedicated PR